src/pynxtools/nomad/parser.py

```diff
@@ -383,13 +385,21 @@ def _get_chemical_formulas(self) -> Set[str]:
             if sample.get("atom_types__field") is not None:
                 atom_types = sample.atom_types__field
                 if isinstance(atom_types, list):
-                    atomlist = atom_types
+                    for symbol in atom_types:
+                        if symbol in chemical_symbols[1::]:
+                            # chemical_symbol from ase.data is ['X', 'H', 'He', ...]
+                            # but 'X' is not a valid chemical symbol just trick to
+                            # have array indices matching element number
+                            element_set.add(symbol)
                 else:
-                    atomlist = atom_types.replace(" ", "").split(",")
-                    chemical_formulas.add("".join(list(set(atomlist))))
+                    for symbol in atom_types.replace(" ", "").split(","):
+                        if symbol in chemical_symbols[1::]:
+                            element_set.add(symbol)
+                    if len(element_set) == 1:
+                        chemical_formulas.add("".join(list(element_set)))
                 # Caution: The element list will be overwritten
                 # in case a single chemical formula is found
-                material.elements = list(set(material.elements) | set(atomlist))
+                material.elements = list(set(material.elements) | element_set)
             if sample.get("chemical_formula__field") is not None:
                 chemical_formulas.add(sample.chemical_formula__field)
```

Comment:
The periodic table in NOMAD should be populated from material.elements, so we should not normalise element_set to checmical_formulas because element set does not deal with the ratio of the elements in the formula and can be misinterpreted.

Reply:
This is what 6d18d6d implemented, in the case that there is really only one element the mapping is possible.
However, I agree with you that even using this special case can be misleading because:
I somebody knows in advance (compared to ones upload to NOMAD) that I populate a field of concept atom_types one agrees that one only indeed lists the elements and thus also accepts that composition information is not recoverable.
Populating the chemical_formula even with a single element can still be factually wrong because NOMAD will pick up on the fact that there is a chemical_formula with in this case one element and concludes that the material is made fully of that element but is not necessarily a correct conclusion that should be NOMADs authority to make.

Reply:
Exactly. Conclusions, and (respective implementation of) normalisation and visualisation functions in NOMAD shall be fixed, too.